File: packages/pigeon/example/app/pigeons/native_interop_example.dart
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:pigeon/pigeon.dart';

// #docregion config
@ConfigurePigeon(
  PigeonOptions(
    dartOptions: DartOptions(),
    kotlinOptions: KotlinOptions(useJni: true),
    swiftOptions: SwiftOptions(useFfi: true, ffiModuleName: 'my_plugin'),
  ),
)
// #enddocregion config
@HostApi()
abstract class NativeInteropExampleApi {
  void doSomething();
}

File: packages/pigeon/example/app/android/app/src/main/kotlin/dev/flutter/pigeon_example_app/NativeInteropExample.kt
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package dev.flutter.pigeon_example_app

// #docregion concurrency-style
suspend fun echoAsync(value: String): String {
  return value
}
// #enddocregion concurrency-style
